Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -149,9 +192,140 @@ export class ArtifactEngine {
         }
     }
 
-    private artifactItemStore: ArtifactItemStore = new ArtifactItemStore();
+    createNewHashMap(sourceProvider: models.IArtifactProvider, itemsToProcess: models.ArtifactItem[]): Promise<string> {
+        return new Promise((resolve,reject) => {
+            sourceProvider.getArtifactItems(itemsToProcess[0]).then((items: models.ArtifactItem[]) => {
+                sourceProvider.getArtifactItem(items.find(x => x.path === path.join(itemsToProcess[0].path, 'artifact-metadata.csv') )).then((hashStream : NodeJS.ReadableStream) => {                        
+                    var newHashPromise = new Promise((resolve) => {
+                        var newHash = readline.createInterface ({
+                            input : hashStream
+                        });
+                
+                        newHash.on('line', (line) => {
+                            var words = line.split(',');
+                            this.newHashMap[words[0]] = words[1];
+                        });
+            
+                        newHash.on('close', () => {
+                            resolve();
+                        });
+                    });
+                    newHashPromise.then(() => {
+                        this.artifactItemStore.setHashMap(this.newHashMap)
+                        resolve();                                   
+                    });                            
+                }, (err) => {
+                    Logger.logError(err);
```

Comment:
refrain for logging error in artifact engine code, we should just reject in case of failures and lets tasks take a call. you can log info messages if you want.